plugin/pkg/scheduler/core/equivalence_cache.go

```diff
@@ -188,12 +188,29 @@ func (ec *EquivalenceCache) InvalidateCachedPredicateItemForPodAdd(pod *v1.Pod,
 
 	// GeneralPredicates: will always be affected by adding a new pod
 	invalidPredicates := sets.NewString("GeneralPredicates")
+
+	// MaxPDVolumeCountPredicate: we check the volumes of pod to make decision.
+	for _, vol := range pod.Spec.Volumes {
```

Comment:
Why does this need to be added here?

Reply:
I noticed `MaxPDVolumeCountChecker` actually relies on volumes count of **specify Pod** to do predicate. So when a pod with PV is add/delete on a node. The `MaxPDVolumeCountChecker` e-cache of this node should be re-calculated.

Hope I am understanding right :)

Reply:
the volume count checker counts the volumes on pods that are assigned on nodes.  I guess for the case where you create the Pod with NodeName already set, then yes, the predicate should be invalidated.